src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+
+        int userValues = 4;
+
+        String[] arrayOfData = new String[userValues];
+        Map<String, Integer> userData = new HashMap<String, Integer>();
+
+        for (String line : data) {
+            arrayOfData = line.split(" ");
```

Comment:
тут використання Мапи трохи оверінженірінг:
```suggestion
            for (String name : names) {
            int salary = 0;
            for (String line : data) {
               String arrayOfData = line.split(" ");
                LocalDate currentDate = LocalDate.parse(arrayOfData[0], FORMATTER);
                if (name and date condition)) {
                    salary += (Integer.parseInt(arrayOfData[3]) * Integer.parseInt(arrayOfData[2]));
                }
            }
            report.append('\n').append(name).append(" - ").append(salary);
        }
```
Таким чином до циклу ти просто створюєш шапку ріпорту і після повертаєш report.toString();